src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -463,34 +464,32 @@ def _make_requirements_from_install_req(
                 ireq.name,
                 ireq.markers,
             )
-            return []
-        if not ireq.link:
+            yield from ()
```

Comment:
```suggestion
```

This line basically does nothing…?

Reply:
Right, didn't pay sufficient attention in the conversion.